agentendpoint/patch_task.go

```diff
@@ -200,7 +200,16 @@ func (r *patchTask) rebootIfNeeded(ctx context.Context, prePatch bool) error {
 		return nil
 	}
 
-	r.RebootCount++
+	if totalRebootCount := r.PrePatchRebootCount + r.PostPatchRebootCount; totalRebootCount > totalRebootCountLimit {
```

Comment:
Are we get back to the same scenario were we will skip the upgrades, 

I don't think we resetting the state of the counters anywhere, so after 5 pre-patch reboots we will never trigger post-patch reboot, am i get it right ?